### PR TITLE
XRT-687 : Dead lock in xocl_drm.c

### PR DESCRIPTION
**Pull request by saifuddin-xilinx**
- Added the missing unlock in xocl_drm.c

### CODE REVIEW

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c [hunk 1]
```diff
@@ -805,11 +805,14 @@ int xocl_init_mem(struct xocl_drm *drm_p)
...
+	if (err) {
+		XOCL_PUT_MEM_TOPOLOGY(drm_p->xdev);
 		return err;
+	}
 
 	if (group_topo == NULL) {
 		err = -ENODEV;
+		XOCL_PUT_MEM_TOPOLOGY(drm_p->xdev);
 		XOCL_PUT_GROUP_TOPOLOGY(drm_p->xdev);
 		return err;
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c:817]: we are still not releasing the lock. add "goto done"

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c [hunk 2]
```diff
@@ -805,11 +805,14 @@ int xocl_init_mem(struct xocl_drm *drm_p)
 
 	/* Initialize memory stats based on Group topology */
 	err = XOCL_GET_GROUP_TOPOLOGY(drm_p->xdev, group_topo);
-	if (err)
+	if (err) {
+		XOCL_PUT_MEM_TOPOLOGY(drm_p->xdev);
```
Comment (chvamshi-xilinx) [inline src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c:809]: we are still not releasing the lock. add "goto done"

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c (hunk 1 above)
Comment (saifuddin-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c:819]: Added

File: src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c (hunk 2 above)
Comment (saifuddin-xilinx) [reply src/runtime_src/core/pcie/driver/linux/xocl/userpf/xocl_drm.c:809]: Added

**Pull request merged**